hr_employee_benefit/models/hr_payslip.py

```diff
@@ -0,0 +1,84 @@
+from odoo import api, fields, models
+
+
+PAYS_PER_YEAR = {
+    'annually': 1,
+    'semi-annually': 2,
+    'quaterly': 4,
+    'bi-monthly': 6,
+    'monthly': 12,
+    'semi-monthly': 24,
+    'bi-weekly': 26,
+    'weekly': 52,
+    'daily': 365,
+}
+
+
+class HrPayslip(models.Model):
+    _inherit = 'hr.payslip'
+
+    benefit_line_ids = fields.One2many(
+        comodel_name='hr.payslip.benefit.line',
+        inverse_name='payslip_id',
+        string='Employee Benefits',
+        readonly=True, states={'draft': [('readonly', False)]},
+    )
+    pays_per_year = fields.Integer(
+        compute='_compute_pays_per_year',
+        string='Number of pays per year', readonly=True,
+        store=True,
+        help="Field required to compute benefits based on an annual "
+        "amount."
+    )
+
+    @api.depends('contract_id')
+    def _compute_pays_per_year(self):
+        self.pays_per_year = PAYS_PER_YEAR.get(
+            self.contract_id.schedule_pay, False)
+
+    @api.multi
+    def _search_benefits(self):
+        """
+        Search employee benefits to be added on the payslip
+
+        This method is meant to be inherited in other modules
+        in order to add benefits from other sources.
+        """
+        self.ensure_one()
+        return self.contract_id.benefit_line_ids
+
+    @api.multi
+    def button_compute_benefits(self):
```

Comment:
you can direct call the compute_benefits method in buttton I think no need to create new method.